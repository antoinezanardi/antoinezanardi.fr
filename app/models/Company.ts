enum CompanyNames {
  SOBOOK = "SoBook",
  OH_MY_CODE = "OhMyCode",
  DAVEO = "Daveo",
}

type Company = {
  name: CompanyNames;
  image: string;
  url: string;
};

const COMPANIES: Record<CompanyNames, Company> = {
  [CompanyNames.OH_MY_CODE]: {
    name: CompanyNames.OH_MY_CODE,
    url: "https://www.sobook.fr/oh-my-code/",
    image: "omc-logo.png",
  },
  [CompanyNames.SOBOOK]: {
    name: CompanyNames.SOBOOK,
    url: "https://www.sobook.fr/oh-my-code/",
    image: "sobook-logo.png",
  },
  [CompanyNames.DAVEO]: {
    name: CompanyNames.DAVEO,
    url: "https://www.daveo.fr/",
    image: "daveo-logo.png",
  },
};

export type { Company };

export { CompanyNames, COMPANIES };