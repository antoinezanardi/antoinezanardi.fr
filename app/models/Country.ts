enum Countries {
  FRANCE = "FRANCE",
  USA = "USA",
  CANADA = "CANADA",
}

export { Countries };