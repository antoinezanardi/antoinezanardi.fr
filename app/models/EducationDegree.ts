import type { School } from "~/models/School";

type Degree = {
  name: string;
  description: string[];
  startedAt?: Date;
  obtainedAt: Date;
};

type EducationDegree = {
  degree: Degree;
  school: School;
};

export type { EducationDegree, Degree };