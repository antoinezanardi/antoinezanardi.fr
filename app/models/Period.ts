class Period {
  public month: number;

  public year: number;

  public constructor(from: Date, to: Date) {
    const firstMonth = 1;
    const monthsInYear = 12;
    this.month = Period.getMonthsDifference(from, to) + firstMonth;
    this.year = Math.floor(this.month / monthsInYear);
    this.month -= this.year * monthsInYear;
  }

  public static getMonthsDifference(from: Date, to: Date): number {
    const monthCountInYear = 12;

    return to.getMonth() - from.getMonth() + monthCountInYear * (to.getFullYear() - from.getFullYear());
  }
}

export { Period };