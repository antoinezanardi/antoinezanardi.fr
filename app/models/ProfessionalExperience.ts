import type { Company } from "~/models/Company";

type Job = {
  name: string;
  description: string[];
  startedAt: Date;
  finishedAt?: Date;
};

type ProfessionalExperience = {
  job: Job;
  company: Company;
};

export type { ProfessionalExperience, Job };