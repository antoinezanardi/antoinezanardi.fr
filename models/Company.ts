enum COMPANY_NAMES {
  SOBOOK = "SoBook",
  OH_MY_CODE = "OhMyCode",
  DAVEO = "Daveo",
}

interface Company {
  name: COMPANY_NAMES;
  image: string;
  url: string;
}

const companies: Record<COMPANY_NAMES, Company> = {
  [COMPANY_NAMES.OH_MY_CODE]: {
    name: COMPANY_NAMES.OH_MY_CODE,
    url: "https://www.sobook.fr/oh-my-code/",
    image: "omc-logo.png",
  },
  [COMPANY_NAMES.SOBOOK]: {
    name: COMPANY_NAMES.SOBOOK,
    url: "https://www.sobook.fr/oh-my-code/",
    image: "sobook-logo.png",
  },
  [COMPANY_NAMES.DAVEO]: {
    name: COMPANY_NAMES.DAVEO,
    url: "https://www.daveo.fr/",
    image: "daveo-logo.png",
  },
};

export type { Company };

export { COMPANY_NAMES, companies };