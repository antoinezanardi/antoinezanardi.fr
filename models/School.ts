enum SCHOOL_NAMES {
  VIMEU = "vimeu",
  EPITECH = "epitech",
  LAVAL = "laval",
  CLOUD_NATIVE_COMPUTED_FOUNDATION = "cloudNativeComputedFoundation",
}

interface School {
  name: SCHOOL_NAMES;
  translatedName?: string;
  city?: string;
  image?: string;
  url: string;
}

const schools: Record<SCHOOL_NAMES, School> = {
  [SCHOOL_NAMES.VIMEU]: {
    name: SCHOOL_NAMES.VIMEU,
    city: "Friville-Escarbotin",
    url: "https://lycee-vimeu-friville.ac-amiens.fr/",
  },
  [SCHOOL_NAMES.EPITECH]: {
    name: SCHOOL_NAMES.EPITECH,
    city: "Lille",
    url: "https://www.epitech.eu/",
    image: "sobook-logo.png",
  },
  [SCHOOL_NAMES.LAVAL]: {
    name: SCHOOL_NAMES.LAVAL,
    city: "Qu√©bec",
    url: "https://www.ulaval.ca/",
    image: "daveo-logo.png",
  },
  [SCHOOL_NAMES.CLOUD_NATIVE_COMPUTED_FOUNDATION]: {
    name: SCHOOL_NAMES.CLOUD_NATIVE_COMPUTED_FOUNDATION,
    url: "https://www.cncf.io/",
    image: "cloud-native-computed-foundation-logo.png",
  },
};

export type { School };

export { SCHOOL_NAMES, schools };