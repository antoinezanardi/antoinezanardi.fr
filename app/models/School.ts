import { Countries } from "~/models/Country";

enum SchoolNames {
  VIMEU = "vimeu",
  EPITECH = "epitech",
  LAVAL = "laval",
  CLOUD_NATIVE_COMPUTING_FOUNDATION = "cloudNativeComputingFoundation",
  GOOGLE = "google",
}

type School = {
  name: SchoolNames;
  translatedName?: string;
  city?: string;
  country: Countries;
  image?: string;
  url: string;
};

const SCHOOLS: Record<SchoolNames, School> = {
  [SchoolNames.VIMEU]: {
    name: SchoolNames.VIMEU,
    city: "Friville-Escarbotin",
    country: Countries.FRANCE,
    url: "https://lycee-vimeu-friville.ac-amiens.fr/",
  },
  [SchoolNames.EPITECH]: {
    name: SchoolNames.EPITECH,
    city: "Lille",
    country: Countries.FRANCE,
    url: "https://www.epitech.eu/",
    image: "epitech-logo.png",
  },
  [SchoolNames.LAVAL]: {
    name: SchoolNames.LAVAL,
    city: "Qu√©bec",
    country: Countries.CANADA,
    url: "https://www.ulaval.ca/",
    image: "laval-logo.png",
  },
  [SchoolNames.CLOUD_NATIVE_COMPUTING_FOUNDATION]: {
    name: SchoolNames.CLOUD_NATIVE_COMPUTING_FOUNDATION,
    country: Countries.USA,
    url: "https://www.cncf.io/",
    image: "cloud-native-computed-foundation-logo.png",
  },
  [SchoolNames.GOOGLE]: {
    name: SchoolNames.GOOGLE,
    country: Countries.USA,
    url: "https://cloud.google.com/",
    image: "google-cloud-platform-logo.png",
  },
};

export type { School };

export { SchoolNames, SCHOOLS };