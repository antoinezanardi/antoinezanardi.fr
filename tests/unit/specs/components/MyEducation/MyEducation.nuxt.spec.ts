import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { MyEducation } from "#components";
import type EducationDegreeCard from "~/components/MyEducation/EducationDegreeCard/EducationDegreeCard.vue";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import type { EducationDegree } from "~/models/EducationDegree";
import { SchoolNames, SCHOOLS } from "~/models/School";

describe("MyEducation Component", () => {
  let wrapper: ReturnType<typeof mount<typeof MyEducation>>;

  async function mountMyEducationComponent(options: ComponentMountingOptions<typeof MyEducation> = {}): Promise<ReturnType<typeof mount<typeof MyEducation>>> {
    return mountSuspendedComponent(MyEducation, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountMyEducationComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Education Degrees", () => {
    it("should render 5 education degree cards when rendered.", () => {
      const educationDegreeCards = wrapper.findAllComponents<typeof EducationDegreeCard>(".education-degree-card");

      expect(educationDegreeCards).toHaveLength(5);
    });

    it("should render GCP ACE education degree card when rendered.", () => {
      const educationDegreeCards = wrapper.findAllComponents<typeof EducationDegreeCard>(".education-degree-card");
      const expectedEducationDegree: EducationDegree = {
        degree: {
          name: `Degrees.gcpACE`,
          description: ["MyEducation.gcpACECertification"],
          obtainedAt: new Date("2022-12-01"),
        },
        school: {
          ...SCHOOLS[SchoolNames.GOOGLE],
          translatedName: `Schools.${SchoolNames.GOOGLE}`,
        },
      };

      expect(educationDegreeCards[0].props("educationDegree")).toStrictEqual<EducationDegree>(expectedEducationDegree);
    });

    it("should render CKAD education degree card when rendered.", () => {
      const educationDegreeCards = wrapper.findAllComponents<typeof EducationDegreeCard>(".education-degree-card");
      const expectedEducationDegree: EducationDegree = {
        degree: {
          name: `Degrees.CKAD`,
          description: ["MyEducation.kubernetesCertification"],
          obtainedAt: new Date("2022-08-01"),
        },
        school: {
          ...SCHOOLS[SchoolNames.CLOUD_NATIVE_COMPUTING_FOUNDATION],
          translatedName: `Schools.${SchoolNames.CLOUD_NATIVE_COMPUTING_FOUNDATION}`,
        },
      };

      expect(educationDegreeCards[1].props("educationDegree")).toStrictEqual<EducationDegree>(expectedEducationDegree);
    });

    it("should render IT Master degree card when rendered.", () => {
      const educationDegreeCards = wrapper.findAllComponents<typeof EducationDegreeCard>(".education-degree-card");
      const expectedEducationDegree: EducationDegree = {
        degree: {
          name: `Degrees.ITMasterDegree`,
          description: ["MyEducation.firstThreeYearsInEpitech", "MyEducation.lastYearsInEpitech"],
          startedAt: new Date("2014-01-01"),
          obtainedAt: new Date("2019-01-01"),
        },
        school: {
          ...SCHOOLS[SchoolNames.EPITECH],
          translatedName: `Schools.${SchoolNames.EPITECH}`,
        },
      };

      expect(educationDegreeCards[2].props("educationDegree")).toStrictEqual<EducationDegree>(expectedEducationDegree);
    });

    it("should render IT certification degree card when rendered.", () => {
      const educationDegreeCards = wrapper.findAllComponents<typeof EducationDegreeCard>(".education-degree-card");
      const expectedEducationDegree: EducationDegree = {
        degree: {
          name: `Degrees.ITCertification`,
          description: ["MyEducation.internationalExperienceForManagement", "MyEducation.softManagementSkills"],
          startedAt: new Date("2017-01-01"),
          obtainedAt: new Date("2018-01-01"),
        },
        school: {
          ...SCHOOLS[SchoolNames.LAVAL],
          translatedName: `Schools.${SchoolNames.LAVAL}`,
        },
      };

      expect(educationDegreeCards[3].props("educationDegree")).toStrictEqual<EducationDegree>(expectedEducationDegree);
    });

    it("should render high school degree card when rendered.", () => {
      const educationDegreeCards = wrapper.findAllComponents<typeof EducationDegreeCard>(".education-degree-card");
      const expectedEducationDegree: EducationDegree = {
        degree: {
          name: `Degrees.highSchoolDiploma`,
          description: ["MyEducation.scientistHighSchoolDiploma"],
          startedAt: new Date("2011-01-01"),
          obtainedAt: new Date("2014-01-01"),
        },
        school: {
          ...SCHOOLS[SchoolNames.VIMEU],
          translatedName: `Schools.${SchoolNames.VIMEU}`,
        },
      };

      expect(educationDegreeCards[4].props("educationDegree")).toStrictEqual<EducationDegree>(expectedEducationDegree);
    });
  });
});