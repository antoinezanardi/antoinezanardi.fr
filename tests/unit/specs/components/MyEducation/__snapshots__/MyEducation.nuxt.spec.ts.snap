// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyEducation Component > should match snapshot when rendered. 1`] = `
"<div id="education" aria-labelledby="education-title" class="section" role="region">
  <div class="cc-education container">
    <section-title-stub icon="fa fa-graduation-cap" title="Formation &amp; Certifications" iconcolor="#F05033" id="education-title"></section-title-stub>
    <education-degree-card-stub educationdegree="[object Object]" aria-label="1 - Degrees.gcpACE" class="education-degree-card" role="region"></education-degree-card-stub>
    <education-degree-card-stub educationdegree="[object Object]" aria-label="2 - Degrees.CKAD" class="education-degree-card" role="region"></education-degree-card-stub>
    <education-degree-card-stub educationdegree="[object Object]" aria-label="3 - Degrees.ITMasterDegree" class="education-degree-card" role="region"></education-degree-card-stub>
    <education-degree-card-stub educationdegree="[object Object]" aria-label="4 - Degrees.ITCertification" class="education-degree-card" role="region"></education-degree-card-stub>
    <education-degree-card-stub educationdegree="[object Object]" aria-label="5 - Degrees.highSchoolDiploma" class="education-degree-card" role="region"></education-degree-card-stub>
  </div>
</div>"
`;
